INTMDB-502-automation code to generate schema file and cfn template with genreri… (#159)
mongodb/mongodbatlas-cloudformation-resources: cfn-resources/autogen/README.md

> # Auto Generate cfn template and generic GO Code

Reviewer: ```suggestion
# Auto Generate CloudFormation template and generic GO Code
```

## How to generate Resource setup and Respective Go template for a new Resource

### Step 1:

Update mapping.json in schema folder with new resource details

> eg:

Reviewer: ```suggestion
For Example:
```

>     `{

Reviewer: Can you double check how this looks in preview? the formatting in off, i belive should be 1 block of code not several lines 

![image](https://user-images.githubusercontent.com/47370943/204931534-7d064e19-70b5-4cdd-b2fa-2e7b46adeaf2.png)

        "resources" : [
            {
                "typeName" :"cloudProviderAccess",
                "openApiPath" : [
                "/api/atlas/v1.0/groups/{groupId}/clusters/{clusterName}/fts/indexes",
                "/api/atlas/v1.0/groups/{groupId}/clusters/{clusterName}/fts/indexes/{databaseName}/{collectionName}",
                "/api/atlas/v1.0/groups/{groupId}/clusters/{clusterName}/fts/indexes/{indexId}"
                ]
            }
        ]
    }`

> ### step 2:

Reviewer: ```suggestion
### Step 2:
```

> ### step 2:

Reviewer: ```suggestion
### Step 2:
```

Run below command to generate schema file that is required for ``cfn generagte`` in the next step.

    make schema

### Step 3:
Run below command to generate Resource template which reads
/configs/<resource>-schema.json and /configs/<resource>-req.json

> eg: to create search-index resource run below command.

Reviewer: ```suggestion
For example: to create search-index resource run below command.
```

search-indexes is the directory name for resource to create
indexes is the ResourceName that can be found in the end of the OpenAPI URL.

> eg: /api/atlas/v1.0/groups/{groupId}/clusters/{clusterName}/fts/indexes

Reviewer: ```suggestion
For Example: /api/atlas/v1.0/groups/{groupId}/clusters/{clusterName}/fts/indexes
```
